src/Features/CSharp/Portable/MakeMethodSynchronous/CSharpMakeMethodSynchronousCodeFixProvider.cs

```diff
@@ -0,0 +1,98 @@
+        {
+            return node.IsKind(SyntaxKind.MethodDeclaration) || node.IsAnyLambdaOrAnonymousMethod();
+        }
+
+        protected override SyntaxNode RemoveAsyncTokenAndFixReturnType(IMethodSymbol methodSymbolOpt, SyntaxNode node)
+        {
+            return node.TypeSwitch(
+                (MethodDeclarationSyntax method) => FixMethod(methodSymbolOpt, method),
+                (AnonymousMethodExpressionSyntax method) => FixAnonymousMethod(method),
+                (ParenthesizedLambdaExpressionSyntax lambda) => FixParenthesizedLambda(lambda),
+                (SimpleLambdaExpressionSyntax lambda) => FixSimpleLambda(lambda),
+                _ => node);
+        }
+
+        private SyntaxNode FixMethod(IMethodSymbol methodSymbol, MethodDeclarationSyntax method)
+        {
+            var newReturnType = method.ReturnType;
+            if (methodSymbol.ReturnType.Name == "Task")
+            {
+                // If the return type is Task<T>, then make the new return type "T".
+                // If it is task, then make the new return type "void".
+                newReturnType = methodSymbol.ReturnType.GetTypeArguments().Length == 1
+                    ? methodSymbol.ReturnType.GetTypeArguments()[0].GenerateTypeSyntax()
+                    : SyntaxFactory.PredefinedType(SyntaxFactory.Token(SyntaxKind.VoidKeyword));
+
+                newReturnType = newReturnType.WithTriviaFrom(method.ReturnType);
+            }
+
+            var asyncTokenIndex = method.Modifiers.IndexOf(SyntaxKind.AsyncKeyword);
+            SyntaxTokenList newModifiers;
+            if (asyncTokenIndex == 0)
+            {
+                // Have to move the trivia on teh async token appropriately.
+                var asyncLeadingTrivia = method.Modifiers[0].LeadingTrivia;
+
+                if (method.Modifiers.Count > 1)
+                {
+                    // Move the trivia to the next modifier;
+                    newModifiers = method.Modifiers.Replace(
+                        method.Modifiers[1],
+                        method.Modifiers[1].WithPrependedLeadingTrivia(asyncLeadingTrivia));
+                    newModifiers = newModifiers.RemoveAt(0);
+                }
+                else
+                {
+                    // move it to the return type.
+                    newModifiers = method.Modifiers.RemoveAt(0);
+                    newReturnType = newReturnType.WithPrependedLeadingTrivia(asyncLeadingTrivia);
+                }
+            }
+            else
+            {
+                newModifiers = method.Modifiers.RemoveAt(asyncTokenIndex);
+            }
+
+            return method.WithReturnType(newReturnType).WithModifiers(newModifiers);
+        }
+
+        private SyntaxNode FixParenthesizedLambda(ParenthesizedLambdaExpressionSyntax lambda)
+        {
+            return lambda.WithAsyncKeyword(default(SyntaxToken)).WithPrependedLeadingTrivia(lambda.AsyncKeyword.LeadingTrivia);
```

Comment:
Put some `/* */` comments around the async tokens and see what happens, or something like:
``` C#
async // Hits the disk somewhere
Task<string> ReadConfig() { }
```

Does the comment survive?